wordpress-mobile/WordPress-iOS
# Automated Development Environment Setup Rake Tasks

This PR adds new tasks to the WordPress iOS rakefile to make it easy for new internal or external developers to get setup with WPiOS quckly.

The new tasks, rake init:oss and rake init:developer, automate much of the setup process that is in the current read me file into a single command that can be run on a new download of WordPress to do the following:
Check for xcode is installed with the correct version
Install all dependencies
install additional development tools
install lint hooks
setup API credentials and generate a filled out wpcom_app_credentials file
And help internal developers create a gpg key for access to production credentials

I have also created a new version of the read me that reflects how to use these new scripts when getting started with WPiOS for the first time.  

To test:
1. Clone a copy of WordPress-ios into a new folder.
2. in terminal go to that directory and run `rake init:oss` or `rake init:developer`  
3. The scripts will run mostly without input 
4. Make sure to have a valid API client App ID and Client secret, you will be prompted to add this info
5.  A new gpg key will also be created in this process (there is an option to skip that)

Once complete make sure that:
The dependencies have been installed and are up to date
the swift link hooks are installed and setup
WordPress ios can compile and log in

PR submission checklist:
